controller/controller.go

```diff
@@ -671,202 +648,144 @@ waitForLock:
 		return err
 	}
 
-	if changedPortRange || MustUpdatePods(&oldConfig, configYAML) {
-		scheduler.NextMajorVersion()
+	shouldRecreatePods := changedPortRange || MustUpdatePods(&oldConfig, configYAML)
+
+	if shouldRecreatePods {
 		l.Info("pods must be recreated, starting process")
+		scheduler.NextMajorVersion()
+	} else {
+		l.Info("pods do not need to be recreated")
+		scheduler.NextMinorVersion()
+	}
 
-		var kubePods *v1.PodList
-		err := mr.WithSegment(models.SegmentPod, func() error {
-			var err error
-			kubePods, err = clientset.CoreV1().Pods(schedulerName).List(metav1.ListOptions{
-				LabelSelector: labels.Set{}.AsSelector().String(),
-				FieldSelector: fields.Everything().String(),
-			})
-			return err
-		})
+	scheduler.RollingUpdateStatus = "in progress"
```

Comment:
RollingUpdateStatus should not be magic strings. Maybe make it an enum (in PG), and use them in code from consts.

Reply:
I decided to leave it as free text. It doesn't has predefined statuses and it is more like a description (should be similar to deis). It has texts like `rolled back to version vXY.Z`.
Also, it should be only human readable.

Reply:
Having some consts would be better than having the strings directly in the code. This way it would at least be easier to avoid typos and have all possible statuses in a single place.

Reply:
I agree.